test/api-client.spec.ts

```diff
@@ -435,15 +432,92 @@ describe('api-client', () => {
     });
 
     describe('getTranscriptObject', () => {
+        const expectedTranscript = {
+                "monologues": [
+                    {
+                        "speaker": 1,
+                        "elements": [
+                            {
+                            "type": "text",
+                            "value": "Hello",
+                            "ts": 0.5,
+                            "end_ts": 1.5,
+                            "confidence": 1
+                            },
+                            {
+                            "type": "text",
+                            "value": "World",
+                            "ts": 1.75,
+                            "end_ts": 2.85,
+                            "confidence": 0.8
+                            },
+                            {
+                            "type": "punct",
+                            "value": "."
+                            }
+                        ]
+                    }
+                ]
+            };
         afterEach(() => {
             expect(mockedAxios.get).toBeCalledWith(`/jobs/${jobId}/transcript`, {
+                responseType: 'json',
                 headers: { 'Accept': `application/vnd.rev.transcript.v1.0+json` }
             });
             expect(mockedAxios.get).toBeCalledTimes(1);
         });
 
-        it('get transcript object', async() => {
-            const expectedTranscript = {
+        it('get transcript object', async () => {
+            const resp = { data: expectedTranscript };
+            mockedAxios.get.mockResolvedValue(resp);
+
+            const transcript = await sut.getTranscriptObject(jobId);
+
+            expect(transcript).toEqual(expectedTranscript);
+        });
+
+        it('handles when api returns unauthorized', async () => {
+            const fakeError = setupFakeApiError(401, "Unauthorized");
```

Comment:
are we repeating these tests for all new endpoint related functions?

Maybe we should extract component which is a wrapper around axios and error handling. Then we can test the wrapper for proper error handling and the api-client only needs to assert that the wrapper was called with the correct params.

Reply:
The more I look at this file the more I think we need to extract the axios wrapper. It will greatly simplify test writing.

Reply:
Talked about this offline. It has been added as an AC for AISDK-49